processor/processor.go

```diff
@@ -494,3 +496,54 @@ func mkRegistry() (*markers.Registry, error) {
 
 	return registry, nil
 }
+
+func parseMarkers(markers markers.MarkerValues) (string, []string) {
+	defaultValue := ""
+	validation := []string{}
+
+	markerNames := make([]string, 0, len(markers))
+	for name := range markers {
+		markerNames = append(markerNames, name)
+	}
+	sort.Strings(markerNames)
+
+	for _, name := range markerNames {
+		value := markers[name][len(markers[name])-1]
+
+		if strings.HasPrefix(name, "kubebuilder:validation:") {
+			name := strings.TrimPrefix(name, "kubebuilder:validation:")
+
+			switch name {
+			case "Pattern":
+				value = fmt.Sprintf("`%s`", value)
+			// FIXME: XValidation currently removed due to being long and difficult to read.
+			// E.g. "XValidation: {self.page < 200 Please start a new book.}"
+			case "XValidation":
+				continue
+			}
+			validation = append(validation, fmt.Sprintf("%s: %v", name, value))
+		}
+
+		if name == "kubebuilder:default" {
+			if value, ok := value.(crdmarkers.Default); ok {
+				defaultValue = fmt.Sprintf("%v", value.Value)
+				if strings.HasPrefix(defaultValue, "map[") {
+					defaultValue = strings.TrimPrefix(defaultValue, "map[")
+					defaultValue = strings.TrimSuffix(defaultValue, "]")
+					defaultValue = fmt.Sprintf("\\{ %s \\}", defaultValue)
```

Comment:
Do we really need to escape curly braces here because backslashes are rendered in asciidoc?

```suggestion
					defaultValue = fmt.Sprintf("{ %s }", defaultValue)
```

IIUC escaping curly braces is to avoid variable substitution, which is the case when there is a string without spaces between the braces.
```asciidoc
:id: 122

/zzaza/{id} --is-rendered--> /zzaza/122

/zzaza/\{id} --is-rendered--> /zzaza/{id}
```

Here, what is between the curly braces is the `key: value` map pair that cannot be substituted?

Escaping:

<img width="2247" alt="Capture d’écran 2024-02-14 à 10 22 25" src="https://github.com/elastic/crd-ref-docs/assets/582883/23bee1c3-d923-4114-8b26-8e73c2e8de65">

No escaping looks fine:

<img width="2247" alt="Capture d’écran 2024-02-14 à 10 22 46" src="https://github.com/elastic/crd-ref-docs/assets/582883/25185cf0-6646-4f23-84e4-b0e5b9bdb26d">

Reply:
We don't in most cases. However, we experienced that the resulting syntax was recognized as something other than text in docusaurus v3. Escaping it appeares to be safe and ensures that its regarded as text everywhere. Can report back with the error shortly.

Reply:
```markdown
# Doc

| Field                | Description | Default      | Validation |
| -------------------- | ----------- | ------------ | ---------- |
| `spec` GuestbookSpec |             | { page:1 }   |            |
| `spec` GuestbookSpec |             | \{ page:1 \} |            |
```

```
Error: MDX compilation failed for file "*****/index.md"
Cause: Could not parse expression with acorn
Details:
{
  "cause": {
    "pos": 186,
    "loc": {
      "line": 5,
      "column": 45
    }
  },
  "column": 46,
  "message": "Could not parse expression with acorn",
  "line": 5,
  "name": "5:46",
  "place": {
    "line": 5,
    "column": 46,
    "offset": 186
  },
  "reason": "Could not parse expression with acorn",
  "ruleId": "acorn",
  "source": "micromark-extension-mdx-expression",
  "url": "https://github.com/micromark/micromark-extension-mdx-expression/tree/main/packages/micromark-extension-mdx-expression#could-not-parse-expression-with-acorn"
}
client (webpack 5.90.1) compiled with 1 error
```

Reply:
Not sure how to best avoid the problem in markdown while not displaying the backslack in asciidoc 🤔 We can for now remove the escape. We'll just do an automatic search replace before building the docusaurs page.

Reply:
If we follow the same path as before for a character escape specific to a given renderer, we could introduce a `RenderDefault` in the markdown renderer. Something along these lines:

```diff
diff --git a/processor/processor.go b/processor/processor.go
index 6bb8904..cc25b64 100644
--- a/processor/processor.go
+++ b/processor/processor.go
@@ -530,7 +530,7 @@ func parseMarkers(markers markers.MarkerValues) (string, []string) {
 				if strings.HasPrefix(defaultValue, "map[") {
 					defaultValue = strings.TrimPrefix(defaultValue, "map[")
 					defaultValue = strings.TrimSuffix(defaultValue, "]")
-					defaultValue = fmt.Sprintf("\\{ %s \\}", defaultValue)
+					defaultValue = fmt.Sprintf("{ %s }", defaultValue)
 				}
 			}
 		}
diff --git a/renderer/markdown.go b/renderer/markdown.go
index ac41bde..34b2f88 100644
--- a/renderer/markdown.go
+++ b/renderer/markdown.go
@@ -79,6 +79,7 @@ func (m *MarkdownRenderer) ToFuncMap() template.FuncMap {
 		"ShouldRenderType":   m.ShouldRenderType,
 		"TypeID":             m.TypeID,
 		"RenderFieldDoc":     m.RenderFieldDoc,
+		"RenderDefault":      m.RenderDefault,
 	}
 }
 
@@ -138,6 +139,13 @@ func (m *MarkdownRenderer) RenderExternalLink(link, text string) string {
 	return fmt.Sprintf("[%s](%s)", text, link)
 }
 
+func (m *MarkdownRenderer) RenderDefault(text string) string {
+	return strings.NewReplacer(
+		"{", "\\{",
+		"}", "\\}",
+	).Replace(text)
+}
+
 func (m *MarkdownRenderer) RenderGVLink(gv types.GroupVersionDetails) string {
 	return m.RenderLocalLink(gv.GroupVersionString())
 }
diff --git a/templates/markdown/type.tpl b/templates/markdown/type.tpl
index e7bd2f1..02ba4e3 100644
--- a/templates/markdown/type.tpl
+++ b/templates/markdown/type.tpl
@@ -31,7 +31,7 @@ _Appears in:_
 {{ end -}}
 
 {{ range $type.Members -}}
-| `{{ .Name  }}` _{{ markdownRenderType .Type }}_ | {{ template "type_members" . }} | {{ .Default }} | {{ range .Validation -}} {{ . }} <br />{{ end }} |
+| `{{ .Name  }}` _{{ markdownRenderType .Type }}_ | {{ template "type_members" . }} | {{ .Default | markdownRenderDefault }} | {{ range .Validation -}} {{ . }} <br />{{ end }} |
 {{ end -}}
 
 {{ end -}}
diff --git a/test/expected.asciidoc b/test/expected.asciidoc
index 69ffe90..d367b2e 100644
--- a/test/expected.asciidoc
+++ b/test/expected.asciidoc
@@ -94,7 +94,7 @@ Guestbook is the Schema for the guestbooks API.
 | *`kind`* __string__ | `Guestbook` | |
 | *`metadata`* __link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#objectmeta-v1-meta[$$ObjectMeta$$]__ | Refer to Kubernetes API documentation for fields of `metadata`.
  |  | 
-| *`spec`* __xref:{anchor_prefix}-github-com-elastic-crd-ref-docs-api-v1-guestbookspec[$$GuestbookSpec$$]__ |  | \{ page:1 \} | 
+| *`spec`* __xref:{anchor_prefix}-github-com-elastic-crd-ref-docs-api-v1-guestbookspec[$$GuestbookSpec$$]__ |  | { page:1 } | 
 |===
 
```

Reply:
Thanks!!